effectful-log-base/src/Effectful/Log.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Effectful.Log
+  ( -- * Logging effect
+    Logging
+  , runLogging
+  , runLoggingWithEnv
+
+    -- * Effectful 'LogBase.MonadLog' methods
+  , Effectful.Log.logMessage
+  , Effectful.Log.localData
+  , Effectful.Log.localDomain
+  , Effectful.Log.getLoggerEnv
+
+    -- * Effectful functions of 'LogBase.Monad'
+  , logMessageEff
+  , getLoggerEff
+  ) where
+
+import Data.Aeson.Types (Pair, Value)
+import Data.Text (Text)
+import Log (Logger, LoggerEnv, LogLevel, MonadLog)
+import qualified Log as LogBase
+import Data.Time.Clock (UTCTime, getCurrentTime)
+
+import Effectful.Internal.Effect
+import Effectful.Internal.Env
+import Effectful.Internal.Monad
+
+-- | An effect for structured logging using the @log-base@ library.
+data Logging :: Effect where
+  Logging :: LoggerEnv -> Logging m r
+
+-- | Run a 'Logging' effect.
+--
+-- This function is the effectful version of 'LogBase.runLogT'.
+runLogging
+  :: IOE :> es
+  => Text
+  -- ^ Application component name to use.
+  -> Logger
+  -- ^ The logging back-end to use.
+  -> Eff (Logging : es) a
+  -- ^ The computation to run.
+  -> Eff es a
+runLogging component logger = runLoggingWithEnv LogBase.LoggerEnv
+  { LogBase.leLogger = logger
+  , LogBase.leComponent = component
+  , LogBase.leDomain = []
+  , LogBase.leData = []
+  }
+
+-- | Run a 'Logging' effect with a given logging environment.
+runLoggingWithEnv
+  :: IOE :> es
+  => LoggerEnv
+  -- ^ The logging environment to use.
+  -> Eff (Logging : es) a
+  -- ^ The computation to run.
+  -> Eff es a
+runLoggingWithEnv logEnv = evalEffect (IdE (Logging logEnv))
+
+----------------------------------------
+-- Effectful 'LogBase.MonadLog' methods
+
+-- | A specialized version of 'LogBase.logMessage'.
+logMessage :: Logging :> es => LogLevel -> Text -> Value -> Eff es ()
+logMessage level message data_ = unsafeEff $ \es -> do
+  IdE (Logging logEnv) <- getEnv es
+  time <- getCurrentTime
```

Comment:
This is not good. `log-base` used to depend on `monad-time` and get its time using `currentTime`, but it was recently changed to use `getCurrentTime` from `Data.Time` so that system time is always used (because `MonadTime` doesn't have to return the current time).

You reintroduced this problem here.